articles/140_topic_and_service_name_mapping.md

```diff
@@ -219,13 +201,94 @@ Since ROS 2 topic and service names are expanded to fully qualified names, any b
 Additionally any URL related syntax, e.g. the `rostopic://` prefix, will be removed once parsed.
 Therefore only forward slashes have to be substituted when converting to DDS topic names.
 
-### Substitution of the Namespace Delimiter
+### DDS Partitions as namespace hierarchy
 
-The namespace delimiter in ROS 2 topic and service names, a forward slash (`/`), will be replaced with double underscores (`__`).
-Note that as fully qualified ROS 2 topic and service names are absolute, there is always a leading forward slash (`/`).
+We chose to implement the ROS namespace hierarchy with the help of DDS Partitions.
+DDS Paritions are part of the Subscription/Publisher QoS settings and have been considered as the prefered method of implementing namespaces.
+A subscriber and publisher only match when not only their basename topic matches, they also have to be specified on the same partiton.
+The big benefit of handling the namespaces with partitions is that it is an official QoS within the DDS specification.
+Furthermore, we can separate a ROS topic hierarchy from a DDS implementation, by forwarding the ROS topic in form of `/ns1/ns2/ns3/base_name` to the DDS implementation, which then takes take about setting up a hierarchy.
+All this stays generally ROS agnostic and open to alternative implementations of the RMW interface (e.g. ZeroMQ etc.), which may want to introduce another, different namespace system.
```

Comment:
I am not sure if I would call this "generally ROS agnostic". Also "introduce another, different namespace system" sounds weird to me.

I would try to formulate it like this: the mapping from ROS names (using a forward slash to separate the namespace and the basename) to DDS partitions and topic names is an implementation details of the RMW implementation. While we use the same mapping for all DDS-based implementations other RMW implementation (e.g. Zero MQ) can decide on a different mapping depending on what the middleware provides.